### ISSUE TITLE
Issue 04: MAE training に gradient clipping と AMP 順序の hardening を追加する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#473]
# Issue 04: MAE training に gradient clipping と AMP 順序の hardening を追加する

## 背景

1000 step pilot の non-finite loss は、stats 異常だけでなく optimizer step 前後の数値不安定性でも起こりうる。Phase 7.5 では、MAE pretraining を長時間実行できるように optional gradient clipping を追加する。

## 目的

`train.grad_clip_norm` を config から指定できるようにし、AMP 有効時は `scaler.unscale_(optimizer)` 後に `torch.nn.utils.clip_grad_norm_()` を呼ぶ。

## 依存 issue

- Issue 03 が先に入っていると、non-finite 発生時の観測性が上がる。ただし実装上の必須依存ではない。

## 変更対象

- 更新: `src/seis_attr_ssl/training/mae.py`
- 更新: `src/seis_attr_ssl/config/validate.py`
- 更新: `proc/configs/mvp_mae.yaml` は optional。デフォルトで有効化するかは Issue 08 でも可。
- 更新: `tests/test_mae_training_engine.py`
- 更新: `tests/test_proc_dry_run.py` または config validation tests

## Config 仕様

`train` に optional key を追加する。

```yaml
train:
  grad_clip_norm: 1.0
```

仕様:

- 未指定または `null` なら clipping しない。
- 指定する場合は finite な正の float/int。
- `bool` は拒否する。
- `0`, 負値、NaN、Inf は拒否する。

`validate_config()` の `_validate_train()` で `grad_clip_norm` を検証する。

## Training loop 仕様

`train_mae_one_epoch()` に optional 引数を追加する。

```python
grad_clip_norm: float | None = None
```

非 AMP の順序:

```python
loss.backward()
if grad_clip_norm is not None:
    grad_norm = torch.nn.utils.clip_grad_norm_(model.parameters(), grad_clip_norm)
optimizer.step()
```

AMP の順序:

```python
scaler.scale(loss).backward()
if grad_clip_norm is not None:
    scaler.unscale_(optimizer)
    grad_norm = torch.nn.utils.clip_grad_norm_(model.parameters(), grad_clip_norm)
scaler.step(optimizer)
scaler.update()
```

`grad_norm` は `MaeTrainingState.metrics` に平均値として含める。key は `grad_norm` とする。clipping 未指定時は metrics に出さなくてよい。

`grad_norm` 自体が non-finite の場合は、可能なら Issue 03 の diagnostic と同じ出力先に gradient summary を出し、`FloatingPointError` で止める。少なくとも silent に optimizer step しないこと。

## 実装メモ

- `model.parameters()` は generator なので、`clip_grad_norm_()` 呼び出し直前に使う。
- AMP disabled の CPU test で動くように CUDA 前提コードを書かない。
- 既存の `amp_enabled = train.amp and cuda available` の仕様は維持する。
- config helper は `training/mae.py` 内にも必要。`_optional_positive_float_config()` 相当を追加する。

## テスト要件

1. `grad_clip_norm` 未指定の既存 tiny training test が通る。
2. `grad_clip_norm = 1.0` で tiny CPU training が通り、checkpoint metrics に `grad_norm` が入る。
3. `grad_clip_norm = 0` または負値で `validate_config()` が失敗する。
4. `grad_clip_norm = True` は型エラーになる。
5. AMP disabled 環境でも `grad_clip_norm` 指定で動く。
6. 可能なら `torch.nn.utils.clip_grad_norm_` を monkeypatch し、AMP disabled path で呼ばれることを確認する。

## 完了条件

- MAE training で optional gradient clipping を使える。
- AMP 時の unscale → clip → step の順序が正しい。
- metrics/checkpoint から grad norm を確認できる。

## 実行コマンド

```bash
python -m pytest tests/test_mae_training_engine.py tests/test_config_loader.py
python -m pytest tests/test_proc_dry_run.py
```

## スコープ外

- optimizer 種別の追加。
- learning rate scheduler。
- resume 実装。これは Issue 06/07。
